src/resources/Search/SearchInterfaces.ts

```diff
@@ -39,6 +39,244 @@ export interface TokenModel {
+    lq?: string;
+    lqPartialMatchKeywords?: string;
+    lqPartialMatchMaxKeywords?: number;
+    lqPartialMatchThreshold?: RestThreshold;
+    maximumAge?: number;
+    mlDidYouMeanMaxCandidates?: number;
+    mlDidYouMeanMinScore?: number;
+    mlDidYouMeanUseFacetCount?: boolean;
+    mlParameters?: RestMLParameters;
+    numberOfResults?: number;
+    parentField?: string;
+    partialMatch?: boolean;
+    partialMatchKeywords?: string;
+    partialMatchThreshold?: RestThreshold;
+    pipeline?: string;
+    q?: string;
+    queryFunctions?: RestQueryFunctionRequest[];
+    questionMark?: boolean;
+    rankingFunctions?: RestRankingFunctionRequest[];
+    recommendation?: string;
+    referrer?: string;
+    resultRankings?: unknown[];
+    retrieveFirstSentences?: boolean;
+    searchById?: boolean;
+    searchHub?: string;
+    sortCriteria?: string;
+    staticQuery?: boolean;
+    summaryLength?: number;
+    syntax?: string;
+    tab?: string;
+    timezone?: string;
+    userActions?: RestUserActionsParameters;
+    visitorId?: string;
+    wildcards?: boolean;
+}
+
+export type RestThreshold = string | number;
+
+export type RestComputedFieldOperation = 'average' | 'maximum' | 'minimum' | 'sum';
+
+export type RestFacetSortCriteria = 'alphanumeric' | 'ascending' | 'descending' | 'occurrences' | 'score';
+
+export type RestFacetType = 'dateRange' | 'hierarchical' | 'numericalRange' | 'specific';
+
+export type RestFormat = 'json' | 'opensearch-atom' | 'opensearch-rss' | 'xlsx';
+
+export type RestGroupBySortCriteria =
+    | 'alphaascending'
+    | 'alphadescending'
+    | 'chisquare'
+    | 'computedfieldascending'
+    | 'computedfielddescending'
+    | 'nosort'
+    | 'occurrences'
+    | 'score';
+
+export type RestMLParametersPadding = 'popular' | 'trending';
+
+export type RestRangeAlgorithm = 'equiprobable' | 'even';
+
+export interface RestActionHistory {
+    internalTime?: number;
+    name?: string;
+    time?: string;
+    value?: string;
+}
+
+export interface RestAnalyticsRequest {
+    clientId?: string;
+    clientRequestId?: string;
+    clientTimestamp?: string;
+    deviceId?: string;
+    documentLocation?: string;
+    documentReferrer?: string;
+    pageId?: string;
+    userAgent?: string;
+    userId?: string;
+    userIp?: string;
+}
+
+export interface RestCategoryFacetRequest {
+    delimitingCharacter?: string;
+    field: string;
+    filterFacetCount?: boolean;
+    injectionDepth?: number;
+    maximumNumberOfValues?: number;
+    path?: string[];
+}
+
+export interface RestComputedField {
+    field: string;
+    operation: RestComputedFieldOperation;
+}
+
+export interface RestContextRequest {
+    [fieldName: string]: string | string[];
+}
+
+export interface RestDictionaryFieldContextRequest {
+    [fieldName: string]: string;
+}
+
+export interface RestFacetOptions {
+    enableIndexFacetOrdering?: boolean;
+    freezeFacetOrder?: boolean;
+}
+
+export interface RestFacetRequest {
```

Comment:
If you want, you could expose the facet request here as a union type. Depending on the type, some facets parameter are just ignored.

Reply:
I could update our documentation so you could create the right types. I could have that by tomorrow